README
${project.name}

${project.description}

All documentation for this plugin can be found at
${project.url}

If you install this plugin manually, make sure it is installed in
lib/plugins/${project.artifactId}/ - if the folder is called different it
will not work!

Please refer to http://www.dokuwiki.org/plugins for additional info
on how to install plugins in DokuWiki.

This project includes parts of OpenLayers.

----
  Copyright (c) 2012-2017 Mark C. Prins <mprins@users.sf.net>

  Permission to use, copy, modify, and distribute this software for any
  purpose with or without fee is hereby granted, provided that the above
  copyright notice and this permission notice appear in all copies.

  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

----
  OpenLayers

  https://raw.github.com/openlayers/openlayers/master/license.txt

Copyright 2005-2017 OpenLayers Contributors. All rights reserved. See
https://raw.github.com/openlayers/openlayers/master/authors.txt for full list.

Redistribution and use in source and binary forms, with or without modification,
are permitted provided that the following conditions are met:

 1. Redistributions of source code must retain the above copyright notice, this
list of conditions and the following disclaimer.

 2. Redistributions in binary form must reproduce the above copyright notice,
this list of conditions and the following disclaimer in the documentation and/or
other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY OPENLAYERS CONTRIBUTORS ``AS IS'' AND ANY EXPRESS
OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
SHALL COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

The views and conclusions contained in the software and documentation are those
of the authors and should not be interpreted as representing official policies,
either expressed or implied, of OpenLayers Contributors.


CHANGES
=== version ${project.version} ===
  * load inline javascript deferred

=== version 4.14 ==
  * made in conjunction with the openlayersmap plugin version ${project.version}

=== version 4.13 ===
  * made in conjunction with the openlayersmap plugin version 4.13

=== version 4.12 ===
  * add support for ArcGIS REST export service

=== version 4.11 ===
  * use latest version of OpenLayers 2.14-dev

=== version 4.10 ===
  * use latest version of OpenLayers 2.14-dev

=== version 4.9 ===
  * some PHP 7 fixes

=== version 4.8 ===
  * -

=== version 4.7 ===
  * PHP 7 compatibility fixes

=== version 4.6 ===
  * use latest version of OpenLayers 2.14-dev

=== version 4.5 ===
  * use latest version of OpenLayers 2.14-dev

=== version 4.4 ===
  * use latest version of OpenLayers 2.14-dev

=== version 4.3 ===
  * use OpenLayers 2.14-dev

=== version 4.2 ===
  * use OpenLayers 2.14-dev
  * sync version number with openlayersmap; as of this version the version numbers should be the same.
  * add support for Mapillary

=== version 1.4 ===
  * use OpenLayers 2.14-dev

=== version 1.3 ===
  * use OpenLayers 2.14-dev

=== version 1.2 ===
  * use OpenLayers 2.13

=== version 1.1 ===
  * provides WMS support

=== version 1.0 ===
  * initial release, provides XYZ support


_test/general.test.php
<?php
/*
 * Copyright (c) 2016 Mark C. Prins <mprins@users.sf.net>
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

/**
 * General tests for the openlayersmapoverlays plugin
 *
 * @group plugin_openlayersmapoverlays
 * @group plugin_dokuwikispatial
 * @group plugins
 */
class general_plugin_openlayersmapoverlays_test extends DokuWikiTest {

    protected $pluginsEnabled = array('openlayersmapoverlays','geophp', 'openlayersmap');

    /**
     * Simple test to make sure the plugin.info.txt is in correct format
     */
    public function test_plugininfo() {
        $file = __DIR__.'/../plugin.info.txt';
        $this->assertFileExists($file);

        $info = confToHash($file);

        $this->assertArrayHasKey('base', $info);
        $this->assertArrayHasKey('author', $info);
        $this->assertArrayHasKey('email', $info);
        $this->assertArrayHasKey('date', $info);
        $this->assertArrayHasKey('name', $info);
        $this->assertArrayHasKey('desc', $info);
        $this->assertArrayHasKey('url', $info);

        $this->assertEquals('openlayersmapoverlays', $info['base']);
        $this->assertRegExp('/^https?:\/\//', $info['url']);
        $this->assertTrue(mail_isvalid($info['email']));
        $this->assertRegExp('/^\d\d\d\d-\d\d-\d\d$/', $info['date']);
        $this->assertTrue(false !== strtotime($info['date']));
    }

    /**
     * test if plugin is loaded.
     */
    public function test_plugin_openlayersmapoverlays_isloaded() {
        global $plugin_controller;
        $this->assertTrue(
                    in_array('geophp', $plugin_controller->getList()),
                    "geophp plugin is loaded"
                            );
        $this->assertTrue(
                    in_array('openlayersmap', $plugin_controller->getList()),
                    "openlayersmap plugin is loaded"
                            );
        $this->assertTrue(
                    in_array('openlayersmapoverlays', $plugin_controller->getList()),
                    "openlayersmapoverlays plugin is loaded"
                            );
    }
}


script.js
/*
 * Copyright (c) 2012-2017 Mark C. Prins <mprins@users.sf.net>
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */

/**
 * add layers to the map based on the olMapOverlays object.
 */
function olovAddToMap() {
	if (olEnable) {
		for ( var key in olMapOverlays) {
			var overlay = olMapOverlays[key];
			var m = olMaps[overlay.id];

			switch (overlay.type) {
			case 'osm':
				m.addLayer(new OpenLayers.Layer.OSM(overlay.name, overlay.url, {
					isBaseLayer : !1,
					opacity : parseFloat(overlay.opacity),
					attribution : overlay.attribution,
					visibility : (overlay.visible).toLowerCase() == 'true',
					tileOptions : {
						crossOriginKeyword : null
					}
				}));
				break;
//			case 'wmts':
//			    var format = new OpenLayers.Format.WMTSCapabilities();
//			    OpenLayers.Request.GET({
//                    url: overlay.url,
//                    params: {
//                        SERVICE: "WMTS",
//                        VERSION: "1.0.0",
//                        REQUEST: "GetCapabilities"
//                    },
//                   success: function(request) {
//                       var doc = request.responseXML;
//                       if (!doc || !doc.documentElement) {
//                           doc = request.responseText;
//                       }
//                       var capabilities = format.read(doc);
//                       var layer = format.createLayer(capabilities, {
//                            name: overlay.name,
//                            layer: overlay.layer,
//                            matrixSet: overlay.matrixSet,
//                            format: overlay.format,
//                            style: overlay.style,
//                            opacity: parseFloat(overlay.opacity),
//                            isBaseLayer: !1
//                       });
//                       map.addLayer(layer);
//                   },
//                   failure: function() {
//                       alert("Trouble getting capabilities doc");
//                       OpenLayers.Console.error.apply(OpenLayers.Console, arguments);
//                   }
//               });

//				m.addLayer(new OpenLayers.Layer.WMTS({
//				    name: overlay.name,
//				    url: overlay.url,
//				    layer: overlay.layer,
//				    style: overlay.style,
//					isBaseLayer: !1,
//					matrixSet: overlay.matrixSet,
//					//matrixIds: overlay.matrixIds,
//					format: overlay.format,
//					opacity : parseFloat(overlay.opacity),
//					attribution : overlay.attribution,
//					visibility : (overlay.visible).toLowerCase() == 'true',
//					tileOptions : {
//						crossOriginKeyword : null
//					}
//				}));
				break;
			case 'wms':
				m.addLayer(new OpenLayers.Layer.WMS(overlay.name, overlay.url, {
					layers : overlay.layers,
					version : overlay.version,
					transparent : overlay.transparent,
					format : overlay.format
				}, {
					opacity : parseFloat(overlay.opacity),
					visibility : (overlay.visible).toLowerCase() == 'true',
					isBaseLayer : !1,
					attribution : overlay.attribution
				}));
				break;
			case 'ags':
				m.addLayer(new OpenLayers.Layer.ArcGIS93Rest(overlay.name, overlay.url, {
					layers : overlay.layers,
					transparent : overlay.transparent,
					format : overlay.format
				}, {
					opacity : parseFloat(overlay.opacity),
					visibility : (overlay.visible).toLowerCase() == 'true',
					isBaseLayer : !1,
					attribution : overlay.attribution
				}));
				break;
			case 'mapillary':
				var mUrl = 'http://api.mapillary.com/v1/im/search?';
				if (overlay.skey !== '') {
					mUrl = 'http://api.mapillary.com/v1/im/sequence?';
				}
				var mLyr = new OpenLayers.Layer.Vector(
						"Mapillary", {
							    projection : new OpenLayers.Projection("EPSG:4326"),
							    strategies : [ new OpenLayers.Strategy.BBOX({
								ratio : 1.1,
								resFactor : 1.5
							}) /* ,new OpenLayers.Strategy.Cluster({}) */],
							protocol : new OpenLayers.Protocol.HTTP({
								url : mUrl,
								format : new OpenLayers.Format.GeoJSON(),
								params : {
									// default to max. 250 locations
									'max-results' : 250,
									'geojson' : true,
									'skey' : overlay.skey
								},
								filterToParams : function(filter, params) {
									if (filter.type === OpenLayers.Filter.Spatial.BBOX) {
										// override the bbox serialization of
										// the filter to give the Mapillary
										// specific bounds
										params['min-lat'] = filter.value.bottom;
										params['max-lat'] = filter.value.top;
										params['min-lon'] = filter.value.left;
										params['max-lon'] = filter.value.right;
										// if the width of our bbox width is
										// less than 0.15 degrees drop the max
										// results
										if (filter.value.top - filter.value.bottom < .15) {
											OpenLayers.Console.debug('dropping max-results parameter, width is: ',
													filter.value.top - filter.value.bottom);
											params['max-results'] = null;
										}
									}
									return params;
								}
							}),
							styleMap : new OpenLayers.StyleMap({
								'default' : {
									cursor : 'help',
									rotation : '${ca}',
									externalGraphic : DOKU_BASE
											+ 'lib/plugins/openlayersmapoverlays/icons/arrow-up-20.png',
									graphicHeight : 20,
									graphicWidth : 20,
								},
								'select' : {
									externalGraphic : DOKU_BASE
											+ 'lib/plugins/openlayersmapoverlays/icons/arrow-up-20-select.png',
									label : '${location}',
									fontSize : '1em',
									fontFamily : 'monospace',
									labelXOffset : '0.5',
									labelYOffset : '0.5',
									labelAlign : 'lb',
								}
							}),
							attribution : '<a href="http://www.mapillary.com/legal.html"><img src="http://mapillary.com/favicon.ico" alt="Mapillary" height="16" width="16" />Mapillary (CC-BY-SA)',
							visibility : (overlay.visible).toLowerCase() == 'true',
						});
				m.addLayer(mLyr);
				selectControl.addLayer(mLyr);
				break;
			case 'search':
				m.addLayer(new OpenLayers.Layer.Vector(
					overlay.name,
					overlay.url,
					{
                        layers : overlay.layers,
                        version : overlay.version,
                        transparent : overlay.transparent,
                        format : overlay.format
					}, {
                        opacity : parseFloat(overlay.opacity),
                        visibility : (overlay.visible).toLowerCase() == 'true',
                        isBaseLayer : !1,
                        attribution : overlay.attribution
					}
				));
				break;
			}
		}
	}
}

var olMapOverlays = new Object();

jQuery(olovAddToMap);


syntax.php
<?php
/*
 * Copyright (c) 2012 Mark C. Prins <mprins@users.sf.net>
*
* Permission to use, copy, modify, and distribute this software for any
* purpose with or without fee is hereby granted, provided that the above
* copyright notice and this permission notice appear in all copies.
*
* THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
* WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
* MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
* ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
* WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
* ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
* OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
*/


// must be run within Dokuwiki
if (!defined('DOKU_INC')) die();

if (!defined('DOKU_LF')) define('DOKU_LF', "\n");
if (!defined('DOKU_TAB')) define('DOKU_TAB', "\t");
if (!defined('DOKU_PLUGIN')) define('DOKU_PLUGIN',DOKU_INC.'lib/plugins/');

require_once DOKU_PLUGIN.'syntax.php';

class syntax_plugin_overlayer extends DokuWiki_Syntax_Plugin {
	public function getType() {
		//return 'FIXME: container|baseonly|formatting|substition|protected|disabled|paragraphs';
		return 'substition';
	}

	public function getSort() {
		return 902;
	}

	public function connectTo($mode) {
		//look for: <overlayer id="olmap" name="sport" url="http://tiles.openseamap.org/sport/${z}/${x}/${y}.png" visible="false"></overlayer>
		$this->Lexer->addSpecialPattern('<overlayer ?[^>\n]*>.*?</overlayer>', $mode, 'plugin_overlayer');
	}

	public function handle($match, $state, $pos, &$handler){
		$param = array ();
		$data = array ();

		preg_match_all('/(\w*)="(.*?)"/us', $match, $param, PREG_SET_ORDER);

		foreach ($param as $kvpair) {
			list ($matched, $key, $val) = $kvpair;
			$key = strtolower($key);
			$data[$key] = strtolower($val);
		}
		return $data;
	}

	public function render($mode, &$renderer, $data) {
		if($mode != 'xhtml') return false;

		list ($id, $url, $name, $visible) = $data;
		$renderer->doc .="\n<script type='text/javascript'><!--//--><![CDATA[//><!--\n";
		$renderer->doc .="olMapOverlays['".$data['id']."'] = {'id':'".$data['id']."','url':'".$data['url']."', 'name':'".$data['name']."', 'visible':'".$data['visible']."'};\n//--><!]]></script>";

		return true;
	}
}


syntax/agslayer.php
<?php
/*
 * Copyright (c) 2017-2020 Mark C. Prins <mprins@users.sf.net>
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */
if (! defined ( 'DOKU_INC' ))
	die ();

if (! defined ( 'DOKU_LF' ))
	define ( 'DOKU_LF', "\n" );
if (! defined ( 'DOKU_TAB' ))
	define ( 'DOKU_TAB', "\t" );
if (! defined ( 'DOKU_PLUGIN' ))
	define ( 'DOKU_PLUGIN', DOKU_INC . 'lib/plugins/' );

require_once DOKU_PLUGIN . 'syntax.php';
/**
 * adds a AGS layer to your map.
 */
class syntax_plugin_openlayersmapoverlays_agslayer extends DokuWiki_Syntax_Plugin {
	private $dflt = array (
			'id' => 'olmap',
			'name' => '',
			'url' => '',
			'opacity' => 0.8,
			'attribution' => '',
			'visible' => false,
			'layers' => '',
			'format' => 'png',
			'transparent' => 'true'
	);

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::getPType()
	 */
	public function getPType() {
		return 'block';
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::getType()
	 */
	public function getType() {
		// return 'FIXME: container|baseonly|formatting|substition|protected|disabled|paragraphs';
		return 'baseonly';
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see Doku_Parser_Mode::getSort()
	 */
	public function getSort() {
		return 904;
	}

	/**
	 * Connect to our special pattern.
	 *
	 * @see Doku_Parser_Mode::connectTo()
	 */
	public function connectTo($mode) {
		// look for: <olmap_agslayer id="olmap" name="cloud" url="http://geoservices2.wallonie.be/arcgis/rest/services/APP_KAYAK/KAYAK/MapServer/export" attribution="wallonie.be" visible="true" layers="show:0,1,2,3,4,7"></olmap_agslayer>
        // sample: http://geoservices2.wallonie.be/arcgis/rest/services/APP_KAYAK/KAYAK/MapServer/export?LAYERS=show%3A0%2C1%2C2%2C3%2C4%2C7&TRANSPARENT=true&FORMAT=png&BBOX=643294.029959%2C6467184.088252%2C645740.014863%2C6469630.073157&SIZE=256%2C256&F=html&BBOXSR=3857&IMAGESR=3857
		$this->Lexer->addSpecialPattern ( '<olmap_agslayer ?[^>\n]*>.*?</olmap_agslayer>',
				$mode, 'plugin_openlayersmapoverlays_agslayer' );
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::handle()
	 */
	public function handle($match, $state, $pos, Doku_Handler $handler) {
		$param = array ();
		$data = $this->dflt;

		preg_match_all ( '/(\w*)="(.*?)"/us', $match, $param, PREG_SET_ORDER );

		foreach ( $param as $kvpair ) {
			list ( $matched, $key, $val ) = $kvpair;
			if (isset ( $data [$key] )) {
				$key = strtolower ( $key );
				$data [$key] = $val;
			}
		}
		return $data;
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::render()
	 */
	public function render($mode, Doku_Renderer $renderer, $data) {
		if ($mode != 'xhtml')
			return false;

		static $loadedOLlib = false;
		if (! $loadedOLlib) {
			$renderer->doc .= DOKU_LF . '<script charset="utf-8" defer="defer" src="' . DOKU_BASE . 'lib/plugins/openlayersmapoverlays/lib/layers.js' . '"></script>';
			$loadedOLlib = true;
		}
		// incremented for each olmap_agslayer tag in the page source
		static $overlaynumber = 0;

		list ( $id, $url, $name, $visible ) = $data;
		$renderer->doc .= DOKU_LF . '<script charset="utf-8" defer="defer" src="data:text/javascript;base64,';
		$str = '{';
		foreach ( $data as $key => $val ) {
			$str .= "'" . $key . "' : '" . $val . "',";
		}
		$str .= "'type':'ags'}";
		$renderer->doc .= base64_encode("olMapOverlays['ags" . $overlaynumber . "'] = " . $str . ";") . '"></script>';
		$overlaynumber ++;
		return true;
	}
}


syntax/osmlayer.php
<?php
/*
 * Copyright (c) 2012-2020 Mark C. Prins <mprins@users.sf.net>
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */
if (! defined ( 'DOKU_INC' ))
	die ();

if (! defined ( 'DOKU_LF' ))
	define ( 'DOKU_LF', "\n" );
if (! defined ( 'DOKU_TAB' ))
	define ( 'DOKU_TAB', "\t" );
if (! defined ( 'DOKU_PLUGIN' ))
	define ( 'DOKU_PLUGIN', DOKU_INC . 'lib/plugins/' );

require_once DOKU_PLUGIN . 'syntax.php';
/**
 * Add OSM style layer to your map.
 */
class syntax_plugin_openlayersmapoverlays_osmlayer extends DokuWiki_Syntax_Plugin {
	private $dflt = array (
			'id' => 'olmap',
			'name' => '',
			'url' => '',
			'opacity' => 0.8,
			'attribution' => '',
			'visible' => false,
			'cors' => null
	);

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::getPType()
	 */
	public function getPType() {
		return 'block';
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::getType()
	 */
	public function getType() {
		// return 'FIXME: container|baseonly|formatting|substition|protected|disabled|paragraphs';
		return 'baseonly';
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see Doku_Parser_Mode::getSort()
	 */
	public function getSort() {
		return 902;
	}

	/**
	 * Connect to our special pattern.
	 *
	 * @see Doku_Parser_Mode::connectTo()
	 */
	public function connectTo($mode) {
		// look for: <olmap_osmlayer id="olmap" name="sport" url="http://tiles.openseamap.org/sport/${z}/${x}/${y}.png" visible="false" opacity=0.6 attribution="Some attribution"></olmap_osmlayer>
		$this->Lexer->addSpecialPattern ( '<olmap_osmlayer ?[^>\n]*>.*?</olmap_osmlayer>',
				$mode, 'plugin_openlayersmapoverlays_osmlayer' );
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::handle()
	 */
	public function handle($match, $state, $pos, Doku_Handler $handler) {
		$param = array ();
		$data = $this->dflt;

		preg_match_all ( '/(\w*)="(.*?)"/us', $match, $param, PREG_SET_ORDER );

		foreach ( $param as $kvpair ) {
			list ( $matched, $key, $val ) = $kvpair;
			if (isset ( $data [$key] )) {
				$key = strtolower ( $key );
				$data [$key] = $val;
			}
		}
		// dbglog($data,'syntax_plugin_overlayer::handle: parsed data is:');
		return $data;
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::render()
	 */
	public function render($mode, Doku_Renderer $renderer, $data) {
		if ($mode != 'xhtml')
			return false;

		// incremented for each olmap_osmlayer tag in the page source
		static $overlaynumber = 0;

		list ( $id, $url, $name, $visible ) = $data;
		$renderer->doc .= DOKU_LF . '<script charset="utf-8" defer="defer" src="data:text/javascript;base64,';
		$str = '{';
		foreach ( $data as $key => $val ) {
			$str .= "'" . $key . "' : '" . $val . "',";
		}
		$str .= '"type":"osm"}';
		$renderer->doc .= base64_encode("olMapOverlays['osm" . $overlaynumber . "'] = " . $str . ";") . '"></script>';
		$overlaynumber ++;
		return true;
	}
}


syntax/wmslayer.php
<?php
/*
 * Copyright (c) 2012-2020 Mark C. Prins <mprins@users.sf.net>
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */
if (! defined ( 'DOKU_INC' ))
	die ();

if (! defined ( 'DOKU_LF' ))
	define ( 'DOKU_LF', "\n" );
if (! defined ( 'DOKU_TAB' ))
	define ( 'DOKU_TAB', "\t" );
if (! defined ( 'DOKU_PLUGIN' ))
	define ( 'DOKU_PLUGIN', DOKU_INC . 'lib/plugins/' );

require_once DOKU_PLUGIN . 'syntax.php';
/**
 * adds a WMS 1.1.1 layer to your map.
 */
class syntax_plugin_openlayersmapoverlays_wmslayer extends DokuWiki_Syntax_Plugin {
	private $dflt = array (
			'id' => 'olmap',
			'name' => '',
			'url' => '',
			'opacity' => 0.8,
			'attribution' => '',
			'visible' => false,
			'layers' => '',
			'version' => '1.1.1',
			'format' => 'image/png',
			'transparent' => 'true'
	);

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::getPType()
	 */
	public function getPType() {
		return 'block';
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::getType()
	 */
	public function getType() {
		// return 'FIXME: container|baseonly|formatting|substition|protected|disabled|paragraphs';
		return 'baseonly';
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see Doku_Parser_Mode::getSort()
	 */
	public function getSort() {
		return 902;
	}

	/**
	 * Connect to our special pattern.
	 *
	 * @see Doku_Parser_Mode::connectTo()
	 */
	public function connectTo($mode) {
		// look for: <olmap_wmslayer id="olmap" name="cloud" url="http://openweathermap.org/t/tile.cgi?SERVICE=WMS" attribution="OpenWeatherMap" visible="true" layers="GLBETA_PR"></olmap_wmslayer>
		$this->Lexer->addSpecialPattern ( '<olmap_wmslayer ?[^>\n]*>.*?</olmap_wmslayer>',
				$mode, 'plugin_openlayersmapoverlays_wmslayer' );
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::handle()
	 */
	public function handle($match, $state, $pos, Doku_Handler $handler) {
		$param = array ();
		$data = $this->dflt;

		preg_match_all ( '/(\w*)="(.*?)"/us', $match, $param, PREG_SET_ORDER );

		foreach ( $param as $kvpair ) {
			list ( $matched, $key, $val ) = $kvpair;
			if (isset ( $data [$key] )) {
				$key = strtolower ( $key );
				$data [$key] = $val;
			}
		}
		// dbglog($data,'syntax_plugin_overlayer::handle: parsed data is:');
		return $data;
	}

	/**
	 * (non-PHPdoc)
	 *
	 * @see DokuWiki_Syntax_Plugin::render()
	 */
	public function render($mode, Doku_Renderer $renderer, $data) {
		if ($mode != 'xhtml')
			return false;

		static $loadedOLlib = false;
		if (! $loadedOLlib) {
			$renderer->doc .= DOKU_LF . '<script charset="utf-8" defer="defer" src="' . DOKU_BASE . 'lib/plugins/openlayersmapoverlays/lib/layers.js' . '"></script>';
			$loadedOLlib = true;
		}
		// incremented for each olmap_wmslayer tag in the page source
		static $overlaynumber = 0;

		list ( $id, $url, $name, $visible ) = $data;
		$renderer->doc .= DOKU_LF . '<script charset="utf-8" defer="defer" src="data:text/javascript;base64,';
		$str = '{';
		foreach ( $data as $key => $val ) {
			$str .= "'" . $key . "' : '" . $val . "',";
		}
		$str .= "'type':'wms'}";
		$renderer->doc .= base64_encode("olMapOverlays['wms" . $overlaynumber . "'] = " . $str . ";") . '"></script>';
		$overlaynumber ++;
		return true;
	}
}